### ISSUE TITLE
The most basic thing does not work 

### ISSUE THREAD

**Issue by felix-tracxpoint** [pyserial/pyserial#334]

**Comment by felix-tracxpoint** [comment]
I installed the pyserial package via pip but when running the most basic piece of code I get a weird error:

```
import serial
s = serial.Serial("COM4",9000)
```

> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-17-156eeea257c8> in <module>()
>       1 import serial
> ----> 2 s = serial.Serial("COM4",9000)
> 
> AttributeError: module 'serial' has no attribute 'Serial'

I tried to hack it a bit and ran this code:

```
from serial import serialwin32 as s32
s = s32.Serial("COM4",9000)
```

But it only led to a more intricate error:

> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-18-12fb96639e59> in <module>()
>       1 from serial import serialwin32 as s32
>       2 print(dir(s32))
> ----> 3 s = s32.Serial("COM4",9000)
> 
> ~\Anaconda3\envs\tracx\lib\site-packages\serial\serialwin32.py in __init__(self, *args, **kwargs)
>      29         self._overlapped_read = None
>      30         self._overlapped_write = None
> ---> 31         super(Serial, self).__init__(*args, **kwargs)
>      32 
>      33     def open(self):
> 
> ~\Anaconda3\envs\tracx\lib\site-packages\serial\serialutil.py in __init__(self, port, baudrate, bytesize, parity, stopbits, timeout, xonxoff, rtscts, write_timeout, dsrdtr, inter_byte_timeout, exclusive, **kwargs)
>     227         self.dsrdtr = dsrdtr
>     228         self.inter_byte_timeout = inter_byte_timeout
> --> 229         self.exclusive = exclusive
>     230 
>     231         # watch for backward compatible kwargs
> 
> ~\Anaconda3\envs\tracx\lib\site-packages\serial\serialwin32.py in exclusive(self, exclusive)
>     473             raise ValueError('win32 only supports exclusive access (not: {})'.format(exclusive))
>     474         else:
> --> 475             serial.SerialBase.exclusive.__set__(self, exclusive)
> 
> AttributeError: module 'serial' has no attribute 'SerialBase'

The port itself is presumably fine since: (i) I am getting information over it via Tera Term when I connect an RTLS device (ii) the test utility worked ok in command line:

> (tracx) C:\Users\USER>python -m serial.tools.list_ports
> COM4
> 1 ports found

What am I doing wrong?

**Comment by zsquareplusc** [comment]
have you named your own module with your test code `serial.py`? Or have any other library installed that provides a module named `serial`?

**Comment by felix-tracxpoint** [comment]
No... In fact, there is no module at all, I am just writing in a Jupyter notebook at this stage and it's called Untitled2. 

As for the second question - is there a way to make sure? 

Thanks!

**Comment by felix-tracxpoint** [comment]
It says:

> AttributeError: module 'serial' has no attribute '__file__'

which is strange because my site-packages directory has a "serial" folder. 

Should I run a separate install or something?

Thanks for helping me with this, I have a feeling we are on to the source of the problem here...

**Comment by rjframe** [comment]
`help(serial)` on the REPL will give you information on the module.

I had another `serial` package installed by my system (a JSON serialization library); I wasn't able to find the pip or apt package to remove it properly so I removed the folder from my local dist-packages folder, then re-installed pyserial. Obviously that may break something else depending on whether you use something that needs that package; a virtualenv would be safer.

**Comment by Eiceman** [comment]
Sorry for dragging up a dead post, but just for anyone in the future, I had a similar issue on a Window 7 machine.  Used pip to uninstall PySerial, but found a hidden file (~yserial) in the site-packages folder.  I removed that manually, started up a fresh command line window, and pip install PySerial --no-cache-dir.  That seemed to work.  Make sure you don't have any other modules named serial.
It seemed like serial.Serial which should be linked to serial.serialwin32 (for the system) wasn't installed properly.  When I tried to call serial.serialwin32 the error I got back said SerialBase wasn't found (though it could be found in the site-packages), so something breaks in the case.
